druid-shell/src/gtk/mod.rs

```diff
@@ -0,0 +1,689 @@
+
+                    state.handler.borrow_mut().mouse_down(
+                        &window::MouseEvent {
+                            pos: Point::from(button.get_position()),
+                            count: get_mouse_click_count(button.get_event_type()),
+                            mods: get_modifiers(button.get_state()),
+                            button: get_mouse_button(button.get_button()),
+                        },
+                        &mut ctx,
+                    );
+                }
+
+                Inhibit(true)
+            });
+        }
+
+        {
+            let handle = handle.clone();
+            drawing_area.connect_button_release_event(move |_widget, button| {
+                if let Some(state) = handle.state.upgrade() {
+                    let mut ctx = WinCtxImpl::from(&handle);
+
+                    state.handler.borrow_mut().mouse_up(
+                        &window::MouseEvent {
+                            pos: Point::from(button.get_position()),
+                            mods: get_modifiers(button.get_state()),
+                            count: 0,
+                            button: get_mouse_button(button.get_button()),
+                        },
+                        &mut ctx,
+                    );
+                }
+
+                Inhibit(true)
+            });
+        }
+
+        {
+            let handle = handle.clone();
+            drawing_area.connect_motion_notify_event(move |_widget, motion| {
+                if let Some(state) = handle.state.upgrade() {
+                    let mut ctx = WinCtxImpl::from(&handle);
+
+                    let pos = Point::from(motion.get_position());
+                    let mouse_event = window::MouseEvent {
+                        pos,
+                        mods: get_modifiers(motion.get_state()),
+                        count: 0,
+                        button: get_mouse_button_from_modifiers(motion.get_state()),
+                    };
+
+                    state
+                        .handler
+                        .borrow_mut()
+                        .mouse_move(&mouse_event, &mut ctx);
+                }
+
+                Inhibit(true)
+            });
+        }
+
+        {
+            let handle = handle.clone();
+            drawing_area.connect_scroll_event(move |_widget, scroll| {
+                if let Some(state) = handle.state.upgrade() {
+                    let mut ctx = WinCtxImpl::from(&handle);
+
+                    let _deltas = scroll.get_scroll_deltas();
+                    // TODO use these deltas (for smooth scrolling)
+                    let modifiers = get_modifiers(scroll.get_state());
+
+                    // The magic "120"s are from Microsoft's documentation for WM_MOUSEWHEEL.
+                    // They claim that one "tick" on a scroll wheel should be 120 units.
+                    // GTK simply reports the direction
```

Comment:
No, GTK doesn't just report the direction, see:

https://gtk-rs.org/docs/gdk/struct.EventScroll.html#method.get_delta